packages/x-date-pickers-pro/src/internal/hooks/useDesktopRangePicker/useDesktopRangePicker.tsx

```diff
@@ -125,22 +130,19 @@ export const useDesktopRangePicker = <
     ownerState: props,
   });
 
-  const componentsForField: BaseMultiInputFieldProps<DateRange<TDate>, unknown>['components'] = {
-    ...fieldProps.components,
-    TextField: components.TextField,
-    Root: components.FieldRoot,
-    Separator: components.FieldSeparator,
+  const slotsForField: BaseMultiInputFieldProps<DateRange<TDate>, unknown>['slots'] = {
+    ...fieldProps.slots,
+    textField: slots.textField,
+    root: slots.fieldRoot,
+    separator: slots.fieldSeparator,
```

Comment:
In theory you could pass slots to the field in two ways:

```tsx
<DatePicker 
  components={{ textField: TextField }}
/>
```

or 

```tsx
<DatePicker 
  componentsProps={{ field: { components: { textField: TextField }}}}
/>
```

With the current code, the 2nd example don't work because the default `components.textField` will always override `componentsProps.field.components.textField`.
With your suggestion, the 2nd example would work.

I don't see any problem applying your suggestion :+1: